Release 0.3

ppaez opened the pull request:
Hola,

Como que ya era hora... este es un intento de llegar al tan esperado milestone 'Release 0.3' :)  Se incorporan los cambios recientes en la rama master, y se agrega lo necesario para soportar hora además de fecha en las búsquedas; faltan unos todavía algunos cambios que están en proceso, para la hora inicial.

Una duda: ¿Para quienes reciben grandes volúmenes de CFDIs, espeficiar horas y minutos es suficiente? Es decir, ¿Una ventana de 1 minuto es suficientemente pequeña?  ¿Qué pasa si se dejan como constantes segundos inciales a 0 y segundos finales a 59 y se permite especificar horas y minutos solamente?

Commit:
Merge branch 'hotfix-0.2.5' into develop

Se agrega soporte para obtener el detalle de conceptos en los reportes, por ejemplo:

{emisor_rfc}|{serie}{folio}|{noIdentificacion}|{descripcion}|{unidad}|{cantidad}|{valorUnitario}|{importe}

Si agrega cualquier campo del detalle, se omite la validación de los sellos y del SAT

mimex committed:
Merge branch 'hotfix-0.2.6' into develop

ppaez committed:
Empezar la referencia de los módulos

- Habilitar la extensión autodoc
- Crear la sección Referencia

ppaez committed:
Explicar la descarga de facturas del SAT

- Qué código está involucrado
- Los pasos generales

ppaez committed:
Explicar los principales módulos de admin-cfdi

- Por ahora solo los que tienen que ver con
  la descarga de CFDIs

ppaez committed:
Crear DescargaSAT

- Es copia sin refactorizar de _download_sat()
- Queda en pyutil.py como las demás herramientas

ppaez committed:
Usar DescargaSAT en script

- Crear descarga.py como script
  para ejecutar desde consola
- Crear pwd.sample como plantilla
  del archivo de credenciales

ppaez committed:
Crear método _download_sat()

- Alineando con la definición propuesta en
  el issue #25

ppaez committed:
Agregar parámetros opcionales para la línea de comando

- Ayuda con -h
- Se ofrecen valores predeterminados

ppaez committed:
Usar ambos parámetros de línea de comando

- La carpeta destino tiene que ser una ruta
  absoluta.  Si es relativa, el navegador
  la ignora y utiliza su valor predeterminado
  de descargas :(

ppaez committed:
Explicar validate_download_sat()

- Las validaciones que hace
- El diccionario que construye
- La tupla que regresa

ppaez committed:
Explicar los valores de data

ppaez committed:
Correcciones

ppaez committed:
Agregar parámetros opcionales año, mes y día

- Los valores predeterminados igual que
  en la aplicación admincfdi

ppaez committed:
Corrección

ppaez committed:
Agregar parámetros opcionales facturas-emitidas y uuid

- El uso de uuid cambia type_search a 1

ppaez committed:
Completar el mock de la aplicación

ppaez committed:
Agregar parámetro opcional rfc-emisor

ppaez committed:
Renombrar self a app

- Se preserva self para referir a
  _download_sat_month() que se va
  a agregar ahora
- Ya podemos refactorizar _download_sat()
  porque podemos probar funcionalmente
  todas las opciones de búsqueda

ppaez committed:
Crear _download_sat_month()

- Es copia sin refactorizar de _download_sat_month()
- Como método de DescargaSAT

ppaez committed:
Agregar parámetro opcional mes-completo

ppaez committed:
Completar el mock de la aplicación

ppaez committed:
Agrupar el bloque main

ppaez committed:
Mover app afuera de main

- Se hará innecesaria mediante refactorización
  en DescargaSAT

ppaez committed:
Renombrar self a app

ppaez committed:
Utilizar values.Global

- Se usan el atributo SAT, page_init y page_cfdi
- Remover del mock de aplication

ppaez committed:
Utilizar pyutil.Util

- Usar get_dates, get_days, sleep
- Remover del mock de aplication

ppaez committed:
Agregar parámetro opcional download_callback

- El atributo progress es llamado para
  proporcionar valor y máximo a un control
  gráfico de tipo barra de progreso
- Por omisión se envían valor y máximo
  a stdout
- Remover pb del mock de aplication

ppaez committed:
Agregar parámetro opcional status_callback

- El atributo status es llamado para
  enviar mensajes
- Por omisión los mensajes se envían
  a stdout
- Remover del mock de aplication

ppaez committed:
Remover parámetro app

- Remover el mock

ppaez committed:
admincfdi usa DescargaSAT

- Agregar los métodos msg_user() y progress()
  como wrapper a los widgets
- Usar estos métodos en los parámetros opcionales
  status_callback y download_callback

ppaez committed:
Remover descarga de admincfdi

- _download_sat() y _download_sat_month()
  ya están incluidas en pyutil.DescargaSAT

ppaez committed:
Merge branch 'separar-download-sat' into develop

arksega committed:
Fix: organiza xmls con nombre en mayúsculas

También se remueve el módulo glob

arksega committed:
Ignorando temporales de vi/vim

arksega committed:
Merge branch 'organizar_xmls' into develop

ppaez committed:
Cambiar nombre a credenciales.conf

ppaez committed:
Renombrar opción a --mes-completo-por-día

ppaez committed:
Agregar parámetro opcional solicitar-credenciales

ppaez committed:
Solicitar las credenciales

ppaez committed:
Carpeta destino es 'cfdi-descarga'

- En la carpeta del usuario

ppaez committed:
Nombre sin la ruta

ppaez committed:
Simplificar la ayuda

ppaez committed:
Agregar parámetros individuales

- En __init__(), _download_sat() y
  _download_sat_month()
- Se pasan solamente los parámetros requeridos
- Remplazar todos los usos de data en
  _download_sat() y _download_sat_month()
  con el parámetro correspondiente

ppaez committed:
Usar los parámetros

- admincfdi.py: _validate_download_sat()
  usa los nombres de los parámetros,
  _button_download_sat_click() llama con
  **data
- descarga.py: data usa los nombres de
  los parámetros y llama con **data

ppaez committed:
Crear prueba funcional de DescargaSAT

- Cuatro casos de prueba
- Utiliza functional_DescargaSAT.conf para
  adecuar los parámetros y resultados esperados
  de cada prueba al usuario con el que
  se realiza

arksega committed:
Agregando sertup.py

arksega committed:
Poniendo cabeceras

pylint 10/10

arksega committed:
Instalación de dependencias

arksega committed:
Mover pyqrcode y pysimplesoap a setup.py

Ahora estos paquetes estan declarados como dependencias en setup.py
por lo que pueder ser removidos de la raiz del proyecto

arksega committed:
Mover archivos a subcarpeta admincfdi

El proyecto necesita estar organizado de forma que la definición del
modulo no este en la raíz del proyecto, de lo contrario setup.py pone
los archivos en raíz de la carpeta destino.

arksega committed:
Referencias a archivos externos relativas

Las imagenes y layouts son ahora referenciadas con respecto a la
ubicación del modulo, no al path de ejecución.

arksega committed:
Usar nueva estructura del modulo

arksega committed:
Usar setuptools en lugar de distuils

arksega committed:
Comentar la inclusión de dependencias de LibreOffice

Commit:
Merge pull request #30 from arksega/develop

Organizar xml en mayúsculas

Commit:
Merge pull request #32 from ppaez/usabilidad-descarga.py

Usabilidad de descarga.py

arksega committed:
Agregar descarga-cfdi a setup.py

arksega committed:
Actualizar referencias a modulo admincfdi

arksega committed:
Agregar cabecera a descarga

Commit:
Se agrega soporte para generar PDF desde una plantilla CSV

Commit:
Se agrega la cadena original

ppaez committed:
Merge pull request #37 from UniversoLibreMexicoAC/template_csv

Template csv

arksega committed:
Actualizar a 0.2.7

arksega committed:
Revert "Comentar la inclusión de dependencias de LibreOffice"

This reverts commit 608e29d1b71525e27559bd54521bfef5a35bafd9.

arksega committed:
Agregar DescargaSAT a admin-cfdi

arksega committed:
Agregar fpdf a setup.py

arksega committed:
Actualizar referencia a pyutil

arksega committed:
Renombrando cfdi2pdf

arksega committed:
Agregar shebang a cfdi2pdf

arksega committed:
Mover template a la carpeta de modulo

arksega committed:
Agregar dependencia pypng

arksega committed:
Agregar carpeta template a setup.py

arksega committed:
Agregar cfdi2pdf a setup.py como script

ppaez committed:
Merge pull request #39 from arksega/setup

Implementacion de setup.py - develop

Commit:
Agregar archivo contributors

Commit:
Actualizar README

Commit:
Cambiado los derechos de Mauricio Baeza a Python Cabal

Commit:
Modificada la instroducción

Commit:
Corregido el termino CFDI del glosario

Commit:
Se agrega el tema instalación al índice

Commit:
Se inicia el tema: instalación

Commit:
Se valida si hay soporte para LibreOffice

Commit:
Se elimina template.py que usa ReportLab

Commit:
Deshabilitar plantilla ODS en interfaz de usuario, si no hay soporte para LibreOffice

Commit:
Se cambia en la interfaz y en código JSON por CSV

arksega committed:
Merge pull request #40 from UniversoLibreMexicoAC/rev_doc

Rev doc
Revisado en la reunión 2015-04-01

arksega committed:
Merge pull request #41 from UniversoLibreMexicoAC/fix_pyuno

Fix pyuno
Verificado durante la reunión de 2015-04-01

mimex committed:
Merge branch 'hotfix-0.2.7' into develop

ppaez committed:
Borrar admincfdi.log

ppaez committed:
Referir al paquete admincfdi

ppaez committed:
Mover la lectura de las credenciales a setUp()

ppaez committed:
Usar directamente los parámetros

ppaez committed:
Separar el llamado de _download_sat()

- Ahora es necesario instanciar DescargaSAT y
  después invocar _download_sat()
- Al instanciar se pueden proporcionar solo los
  dos callbacks opcionales.
- _download_sat() recibe los demás parámetros
  opcionales.
- Se validan estos cambios con las pruebas
  funcionales.
- El uso de _download_sat() se actualizó en
  las pruebas funcionales.

ppaez committed:
Actualizar las aplicaciones

- admin-cfdi y descarga-cfdi llaman a
  _download_sat() después de instanciar
  DescargaSAT

ppaez committed:
Prueba unitaria previa a refactorizar

- Se desea separar _download_sat() en partes
  más pequeñas.
- Empezando por sacar el bloque que define el
  perfil del navegador a get_firefox_profile()

ppaez committed:
Implementación de get_firefox_profile()

- Reutiliza el código existente

ppaez committed:
Uso de get_firefox_profile()

- El cambio se valida con la prueba funcional

ppaez committed:
Crear DescargaSAT.connect()

- Reutiliza el código existente
- Persiste la referencia a la instancia
  de webbrowser que crea
- Se valida con una prueba funcional.  La sesión
  queda abierta y el navegador no se termina
  porque aún no tenemos DescargaSAT.disconnect()

ppaez committed:
Uso de DescargaSAT.connect()

- El cambio se valida con las pruebas funcionales

ppaez committed:
Crear DescargaSAT.disconnect()

- Reutiliza el código existente
- Se valida con una prueba funcional

ppaez committed:
Uso de DescargaSAT.disconnect()

- El cambio se valida con las pruebas funcionales

ppaez committed:
Crear DescargaSAT.search()

- Reutiliza el código existente,
  cubierto con pruebas unitarias
- Se valida con una prueba funcional

ppaez committed:
Crear DescargaSAT.download()

ppaez committed:
Agregar faltantes de _download_sat()

- Status inicial
- Status de búsqueda sin resultados

ppaez committed:
Actualizar las aplicaciones

- admin-cfdi y  descarga-cfdi dejan
  de usar _download_sat(), se
  remplaza por connect(), search(),
  download() y disconnect()
- Se conserva el try/except/finally
  que asegura la desconexión si hay
  algún problema
- Se actualizan las pruebas funcionales

ppaez committed:
Actualizar las pruebas unitarias

ppaez committed:
Remover _download_sat()

jjlopez committed:
Merge pull request #45 from ppaez/correcciones

Borrar admincfdi.log

jjlopez committed:
Merge pull request #46 from ppaez/refactorizar-DescargaSat

Refactorizar descarga sat

ppaez committed:
Actualizar la referencia a los módulos

ppaez committed:
Comenzar la documentación de la API

- Una sección para cada clase
- Uso de DescargaSAT y ejemplo

ppaez committed:
Documentar el uso de las pruebas funcionales

Commit:
Se reemplaza pysimplesoap por request para verificar estatus en el SAT

Commit:
Merge pull request #48 from ppaez/documentar-api-y-pruebas

Documentar api y pruebas funcionales

ppaez committed:
Merge pull request #50 from UniversoLibreMexicoAC/estatus_sat

Se reemplaza pysimplesoap por request para verificar estatus en el SAT

Commit:
Se selecciona la hora y minuto para busquedas de facturas emitidas

ppaez committed:
Merge pull request #53 from UniversoLibreMexicoAC/bug-facturas-emitidas

Se selecciona la hora y minuto para busquedas de facturas emitidas

Commit:
La validación ahora soporta el complemente terceros

ppaez committed:
Importar módulos

Commit:
Se actualiza el inicio de descarga a 2014, el SAT solo tiene a partir de este año

ppaez committed:
Agregar prueba para facturas emitidas

Commit:
Se agregar soporte para descargar completa del mes para las facturas emitidas

ppaez committed:
Remover el parámetro type_search de search()

- Utilizar uuid
- Actualizar las aplicaciones admin-cfdi
  y descarga-cfdi
- Actualizar los ejemplos en la documentación
  de la API
- Actualizar pruebas unitarias y funcionales

ppaez committed:
Estandarizar facturas_emitidas como booleano

- Actualizar en search()
- Actualizar las aplicaciones admin-cfdi
  y descarga-cfdi
- Actualizar pruebas unitarias y funcionales

ppaez committed:
Remover dos parámetros de search()

- rfc y ciec se usan ya en connect()

ppaez committed:
Usar explicit waits

- connect() cubierto
- search() empezado
- Actualizar pruebas unitarias

ppaez committed:
Año, mes y día en facturas recibidas

ppaez committed:
Esperar al elemento día actualizado

ppaez committed:
Esperar a los resultados

- Una vez hecha la actualización, se prueba
  si existe el elemento de resultados para
  saber si hay facturas

ppaez committed:
Esperar la lista de botones de descarga

- Asegurar que estén actualizados los botones
  y prevenir un error intermitente más adelante
  en download()

ppaez committed:
Merge pull request #55 from UniversoLibreMexicoAC/bug_terceros

La validación ahora soporta el complemente terceros

ppaez committed:
Merge pull request #56 from UniversoLibreMexicoAC/cfdi_emitidas_dia

Cfdi emitidas

ppaez committed:
Cambiar el valor predeterminado de día para search()

- Se requiere un valor que pueda convertirse a entero
  para buscar facturas emitidas

ppaez committed:
Agregar una expected condition a Selenium

- Se tomó y adaptó el código de la función visibility_of()
  en selenium.webdriver.support.expected_conditions
- Se tomó el código de la función auxiliar _element_if_visible()
  del mismo módulo.
- Esta nueva expected condition es necesaria para la espera
  después de que se envía la forma de búsqueda.

ppaez committed:
Usar visibility_of_either()

ppaez committed:
Agregar espera para borrado

- La tabla de resultados se descarta después
  de enviar la forma de búsqueda, antes de
  que esté disponible de nuevo

ppaez committed:
Usar el valor completo

ppaez committed:
Agregar y usar constantes en values.Global.SAT

Commit:
Merge pull request #57 from ppaez/actualizar-api-descarga

Actualizar la API de descarga

Commit:
Merge pull request #58 from ppaez/explicit-waits

Uso de explicit waits

ppaez committed:
Accesar los selects directamente

Commit:
Merge pull request #59 from ppaez/simplificar-search

Accesar los selects directamente

Commit:
Solucion para el iusse 14

ppaez committed:
Separar las aplicaciones de la biblioteca

ppaez committed:
Actualizar los pasos de descarga

ppaez committed:
Primero va pyutil

ppaez committed:
Separar admin-cfdi

ppaez committed:
Mejorar redacción

ppaez committed:
Separar descarga-cfdi

ppaez committed:
Merge pull request #60 from UniversoLibreMexicoAC/iusse_14

Solucion para el iusse 14

ppaez committed:
Accesar los selects directamente

- Hora, minuto y segundo para facturas
  emitidas
- Se usan las constantes ya existentes
  en value.Global.SAT

ppaez committed:
Prevenir error esporádico de elemento inexistente

ppaez committed:
Prevenir timeout en búsqueda de facturas emitidas

- La respuesta del servidor puede tardar más en la búsqueda
  que con facturas recibidas, se aumenta el valor global
  del timeout

ppaez committed:
Mover al capítulo de uso

ppaez committed:
Agregar síntesis de los pasos

ppaez committed:
Agregar detalles de los pasos

ppaez committed:
Explicar paginación

ppaez committed:
Estandarizar

Commit:
Merge pull request #61 from ppaez/actualizar-documentacion

Actualizar documentacion

Commit:
Merge pull request #63 from ppaez/simplificar-emitidas

Simplificar emitidas

ppaez committed:
Valores enviados con las formas de búsqueda

ppaez committed:
Agregar scripts para Windows #67

- admin-cfdi.pyw sin consola
- descarga-cfdi.cmd y cfdi2pdf.cmd en consola
- actualizar setup.py para que los instale
- actualizar documentación

ppaez committed:
Soportar HOMEPATH en Windows #65

ppaez committed:
Corregir error por codepage 437 en consola

- Queda pendiente arreglar acentos y ñ

Commit:
Merge pull request #68 from ppaez/windows-fixes

Windows fixes

Commit:
Merge pull request #69 from ppaez/detalles-descarga-sat

Detalles de descarga del SAT

Commit:
Soporte para generar PDF desde CSV desde GUI

Commit:
Corregida una asignacion de variable al generar PDF desde ODS

ppaez committed:
Merge pull request #70 from UniversoLibreMexicoAC/csv2pdf

Generar PDF desde CSV desde GUI

Commit:
Se agrega permiso de ejecución al archivo admin-cfdi

Commit:
Actualizado gitignore.\nCorregido un error en setup

Commit:
Actualizado README

Commit:
Merge pull request #71 from UniversoLibreMexicoAC/rev-doc

Rev doc

Commit:
FIX - Variable COLOR_RED en pyutil

ppaez committed:
Merge pull request #75 from UniversoLibreMexicoAC/develop

FIX - Variable COLOR_RED en pyutil

ppaez committed:
Crear lee_credenciales()

- Se establece una convención en status para
  simplificar el manejo de errores en las
  aplicaciones: 'Ok' cuando hay éxito, un
  mensaje de error en otros casos.

- Se definen dos casos de error que serían
  los más frecuentes y las pruebas unitarias
  los cubren.

- Si más adelante se implementa la encriptación
  del archivo de credenciales, se puede agregar
  a este nivel sin modificar las aplicaciones.

ppaez committed:
Usar lee_credenciales()

- descarga-cfdi ahora informa al usuario sobre dos
  posibles fallas al leer las credenciales y termina
  con código de salida 1.

- las pruebas funcionales verifican que se cargaron
  las credenciales antes de cada prueba, en caso de
  un error éste se muestra claramente.

ppaez committed:
Agregar prueba

- Asegurar que espacios adicionales y tabulador
  introducidos por el usuario, o fin de línea y
  retorno introducido por el editor no afectan

ppaez committed:
Mejorar la introducción

- Mostrar la ventana principal de admin-cfdi.  A 50% de escala
  pero permite agrandar a 100% haciendo clic.
- Mencionar ambas aplicaciones de línea de comando, mostrando
  un ejemplo de su salida

ppaez committed:
Agregar ejemplo de PDF y mencionar la licencia

Commit:
Merge pull request #78 from ppaez/lee-credenciales

Lee credenciales

Commit:
Fix - Corrección al obtener los decimales

Commit:
Fix - Corrección al obtener los decimales

ppaez committed:
Merge pull request #81 from UniversoLibreMexicoAC/decimales

Fix - Corrección al obtener los decimales

ppaez committed:
Fix - admin-cfdi.pyw

Commit:
Merge pull request #82 from ppaez/develop

Fix - admin-cfdi.pyw

Commit:
Merge pull request #84 from ppaez/organizar-docs

Agregar contenido a la Introducción

Commit:
Quitar acento en html_title en conf.py para que se genere correctamente la documentación

Commit:
Merge pull request #85 from UniversoLibreMexicoAC/develop

Quitar acento en html_title en conf.py para que …

Commit:
Se regresa a 2011 como año de inicio para descargas

ppaez committed:
Merge pull request #86 from UniversoLibreMexicoAC/develop

Se regresa a 2011 como año de inicio para descargas

Commit:
Fix - Ahora se identifica correctamente tanto si la interfaz esta en ingles o en español

ppaez committed:
Merge pull request #88 from UniversoLibreMexicoAC/develop

Fix - Ahora se identifica correctamente …

ppaez committed:
Cobertura para DescargaSAT.connect()

ppaez committed:
Agregar prueba funcional

- Se verifica el valor regresado por
  DescargaSAT.connect() y se cubre el
  caso de falla.

ppaez committed:
Actualizar documentación

Commit:
Merge pull request #89 from ppaez/pruebas-connect

Pruebas connect

ppaez committed:
Merge branch 'UniversoLibre/por-hora' into release-0.3

ppaez committed:
Remover comandos innecesarios

ppaez committed:
Soportar hora y minuto final

- Agregar parámetros opcionales hora_final y
  minuto_final a DescargaSAT.search()
- Se descartan ceros a la izquierda, de lo contrario
  la página del SAT ignora el valor.  Se explica
  en la documentación
- Agregar una prueba funcional
- Agregar ambos parámetros opcionales a descarga-cfdi
- admin-cfdi y descarga-cfdi pasan ambos parámetros
  a DescargaSAT.search()

omar7c1 commented:
El 13 de febrero de 2016, 8:27, Patricio Paez notifications@github.com
escribió:

> Hola,
> 
> Como que ya era hora... este es un intento de llegar al tan esperado
> milestone 'Release 0.3' :) Se incorporan los cambios recientes en la rama
> master, y se agrega lo necesario para soportar hora además de fecha en las
> búsquedas; faltan unos todavía algunos cambios que están en proceso, para
> la hora inicial.
> 
> Una duda: ¿Para quienes reciben grandes volúmenes de CFDIs, espeficiar
> horas y minutos es suficiente? Es decir, ¿Una ventana de 1 minuto es
> suficientemente pequeña? ¿Qué pasa si se dejan como constantes segundos
> inciales a 0 y segundos finales a 59 y se permite especificar horas y
> minutos solamente?
> 
> Nunca es suficiente pero si ayuda, yo e tenido clientes que en 10 min
> llegan mas de 200 facturas. lo ideal es que si el tiene la necesidad pueda
> hacer la descarga en esos horarios pesados por minuto o por 5 minutos
> 
> ---
> 
> You can view, comment on, or merge this pull request online at:
> 
>   https://github.com/LinuxCabal/admin-cfdi/pull/91
> Commit Summary
> - Merge branch 'hotfix-0.2.5' into develop
> - Merge branch 'hotfix-0.2.6' into develop
> - Empezar la referencia de los módulos
> - Explicar la descarga de facturas del SAT
> - Explicar los principales módulos de admin-cfdi
> - Crear DescargaSAT
> - Usar DescargaSAT en script
> - Crear método _download_sat()
> - Agregar parámetros opcionales para la línea de comando
> - Usar ambos parámetros de línea de comando
> - Explicar validate_download_sat()
> - Explicar los valores de data
> - Correcciones
> - Agregar parámetros opcionales año, mes y día
> - Corrección
> - Agregar parámetros opcionales facturas-emitidas y uuid
> - Completar el mock de la aplicación
> - Agregar parámetro opcional rfc-emisor
> - Renombrar self a app
> - Crear _download_sat_month()
> - Agregar parámetro opcional mes-completo
> - Completar el mock de la aplicación
> - Agrupar el bloque main
> - Mover app afuera de main
> - Renombrar self a app
> - Utilizar pyutil.Util
> - Utilizar values.Global
> - Agregar parámetro opcional status_callback
> - Agregar parámetro opcional download_callback
> - Remover parámetro app
> - admincfdi usa DescargaSAT
> - Remover descarga de admincfdi
> - Merge branch 'separar-download-sat' into develop
> - Fix: organiza xmls con nombre en mayúsculas
> - Ignorando temporales de vi/vim
> - Merge branch 'organizar_xmls' into develop
> - Renombrar opción a --mes-completo-por-día
> - Cambiar nombre a credenciales.conf
> - Agregar parámetro opcional solicitar-credenciales
> - Solicitar las credenciales
> - Carpeta destino es 'cfdi-descarga'
> - Nombre sin la ruta
> - Simplificar la ayuda
> - Merge pull request #30 from arksega/develop
> - Merge pull request #32 from ppaez/usabilidad-descarga.py
> - Se agrega soporte para generar PDF desde una plantilla CSV
> - Se agrega la cadena original
> - Merge pull request #37 from UniversoLibreMexicoAC/template_csv
> - Agregando sertup.py
> - Poniendo cabeceras
> - Instalación de dependencias
> - Mover pyqrcode y pysimplesoap a setup.py
> - Mover archivos a subcarpeta admincfdi
> - Referencias a archivos externos relativas
> - Usar nueva estructura del modulo
> - Usar setuptools en lugar de distuils
> - Comentar la inclusión de dependencias de LibreOffice
> - Agregar descarga-cfdi a setup.py
> - Actualizar referencias a modulo admincfdi
> - Agregar cabecera a descarga
> - Actualizar a 0.2.7
> - Revert "Comentar la inclusión de dependencias de LibreOffice"
> - Agregar DescargaSAT a admin-cfdi
> - Agregar fpdf a setup.py
> - Actualizar referencia a pyutil
> - Renombrando cfdi2pdf
> - Agregar shebang a cfdi2pdf
> - Mover template a la carpeta de modulo
> - Agregar dependencia pypng
> - Agregar carpeta template a setup.py
> - Agregar cfdi2pdf a setup.py como script
> - Merge pull request #39 from arksega/setup
> - Agregar archivo contributors
> - Actualizar README
> - Cambiado los derechos de Mauricio Baeza a Python Cabal
> - Modificada la instroducción
> - Corregido el termino CFDI del glosario
> - Se agrega el tema instalación al índice
> - Se inicia el tema: instalación
> - Se valida si hay soporte para LibreOffice
> - Se elimina template.py que usa ReportLab
> - Deshabilitar plantilla ODS en interfaz de usuario, si no hay soporte
>   para LibreOffice
> - Se cambia en la interfaz y en código JSON por CSV
> - Merge pull request #40 from UniversoLibreMexicoAC/rev_doc
> - Merge pull request #41 from UniversoLibreMexicoAC/fix_pyuno
> - Merge branch 'hotfix-0.2.7' into develop
> - Borrar admincfdi.log
> - Agregar parámetros individuales
> - Usar los parámetros
> - Crear prueba funcional de DescargaSAT
> - Referir al paquete admincfdi
> - Mover la lectura de las credenciales a setUp()
> - Usar directamente los parámetros
> - Separar el llamado de _download_sat()
> - Actualizar las aplicaciones
> - Prueba unitaria previa a refactorizar
> - Implementación de get_firefox_profile()
> - Uso de get_firefox_profile()
> - Crear DescargaSAT.connect()
> - Uso de DescargaSAT.connect()
> - Crear DescargaSAT.disconnect()
> - Uso de DescargaSAT.disconnect()
> - Crear DescargaSAT.search()
> - Crear DescargaSAT.download()
> - Agregar faltantes de _download_sat()
> - Actualizar las aplicaciones
> - Actualizar las pruebas unitarias
> - Remover _download_sat()
> - Merge pull request #45 from ppaez/correcciones
> - Merge pull request #46 from ppaez/refactorizar-DescargaSat
> - Actualizar la referencia a los módulos
> - Comenzar la documentación de la API
> - Documentar el uso de las pruebas funcionales
> - Se reemplaza pysimplesoap por request para verificar estatus en el
>   SAT
> - Merge pull request #48 from ppaez/documentar-api-y-pruebas
> - Merge pull request #50 from UniversoLibreMexicoAC/estatus_sat
> - Se selecciona la hora y minuto para busquedas de facturas emitidas
> - Merge pull request #53 from
>   UniversoLibreMexicoAC/bug-facturas-emitidas
> - La validación ahora soporta el complemente terceros
> - Se actualiza el inicio de descarga a 2014, el SAT solo tiene a
>   partir de este año
> - Se agregar soporte para descargar completa del mes para las facturas
>   emitidas
> - Merge pull request #55 from UniversoLibreMexicoAC/bug_terceros
> - Merge pull request #56 from UniversoLibreMexicoAC/cfdi_emitidas_dia
> - Agregar prueba para facturas emitidas
> - Remover el parámetro type_search de search()
> - Estandarizar facturas_emitidas como booleano
> - Remover dos parámetros de search()
> - Cambiar el valor predeterminado de día para search()
> - Importar módulos
> - Usar explicit waits
> - Año, mes y día en facturas recibidas
> - Esperar al elemento día actualizado
> - Esperar a los resultados
> - Esperar la lista de botones de descarga
> - Agregar una expected condition a Selenium
> - Usar visibility_of_either()
> - Agregar espera para borrado
> - Usar el valor completo
> - Agregar y usar constantes en values.Global.SAT
> - Merge pull request #57 from ppaez/actualizar-api-descarga
> - Merge pull request #58 from ppaez/explicit-waits
> - Accesar los selects directamente
> - Merge pull request #59 from ppaez/simplificar-search
> - Solucion para el iusse 14
> - Separar las aplicaciones de la biblioteca
> - Actualizar los pasos de descarga
> - Primero va pyutil
> - Separar admin-cfdi
> - Mejorar redacción
> - Separar descarga-cfdi
> - Merge pull request #60 from UniversoLibreMexicoAC/iusse_14
> - Accesar los selects directamente
> - Prevenir error esporádico de elemento inexistente
> - Prevenir timeout en búsqueda de facturas emitidas
> - Mover al capítulo de uso
> - Agregar síntesis de los pasos
> - Agregar detalles de los pasos
> - Explicar paginación
> - Estandarizar
> - Merge pull request #61 from ppaez/actualizar-documentacion
> - Merge pull request #63 from ppaez/simplificar-emitidas
> - Valores enviados con las formas de búsqueda
> - Agregar scripts para Windows #67
> - Soportar HOMEPATH en Windows #65
> - Corregir error por codepage 437 en consola
> - Merge pull request #68 from ppaez/windows-fixes
> - Merge pull request #69 from ppaez/detalles-descarga-sat
> - Soporte para generar PDF desde CSV desde GUI
> - Corregida una asignacion de variable al generar PDF desde ODS
> - Merge pull request #70 from UniversoLibreMexicoAC/csv2pdf
> - Se agrega permiso de ejecución al archivo admin-cfdi
> - Actualizado gitignore.\nCorregido un error en setup
> - Actualizado README
> - Merge pull request #71 from UniversoLibreMexicoAC/rev-doc
> - FIX - Variable COLOR_RED en pyutil
> - Merge pull request #75 from UniversoLibreMexicoAC/develop
> - Crear lee_credenciales()
> - Usar lee_credenciales()
> - Agregar prueba
> - Merge pull request #78 from ppaez/lee-credenciales
> - Fix - Corrección al obtener los decimales
> - Fix - Corrección al obtener los decimales
> - Merge pull request #81 from UniversoLibreMexicoAC/decimales
> - Fix - admin-cfdi.pyw
> - Merge pull request #82 from ppaez/develop
> - Mejorar la introducción
> - Agregar ejemplo de PDF y mencionar la licencia
> - Merge pull request #84 from ppaez/organizar-docs
> - Quitar acento en html_title en conf.py para que se genere
>   correctamente la documentación
> - Merge pull request #85 from UniversoLibreMexicoAC/develop
> - Se regresa a 2011 como año de inicio para descargas
> - Merge pull request #86 from UniversoLibreMexicoAC/develop
> - Fix - Ahora se identifica correctamente tanto si la interfaz esta en
>   ingles o en español
> - Merge pull request #88 from UniversoLibreMexicoAC/develop
> - Cobertura para DescargaSAT.connect()
> - Agregar prueba funcional
> - Actualizar documentación
> - Merge pull request #89 from ppaez/pruebas-connect
> - Merge branch 'UniversoLibre/por-hora' into release-0.3
> - Remover comandos innecesarios
> - Soportar hora y minuto final
> 
> File Changes
> - _M_ .gitignore
>   https://github.com/LinuxCabal/admin-cfdi/pull/91/files#diff-0 (4)
> - _M_ README.rst
>   https://github.com/LinuxCabal/admin-cfdi/pull/91/files#diff-1 (27)
> - _R_ admin-cfdi
>   https://github.com/LinuxCabal/admin-cfdi/pull/91/files#diff-2 (469)
> - _A_ admin-cfdi.pyw
>   https://github.com/LinuxCabal/admin-cfdi/pull/91/files#diff-3 (6)
> - _D_ admincfdi.log
>   https://github.com/LinuxCabal/admin-cfdi/pull/91/files#diff-4 (0)
> - _R_ admincfdi/**init**.py
>   https://github.com/LinuxCabal/admin-cfdi/pull/91/files#diff-5 (0)
> - _R_ admincfdi/bin/cadena3.2.xslt
>   https://github.com/LinuxCabal/admin-cfdi/pull/91/files#diff-6 (0)
> - _R_ admincfdi/bin/cfdi_3.2.xsd
>   https://github.com/LinuxCabal/admin-cfdi/pull/91/files#diff-7 (0)
> - _R_ admincfdi/bin/cfdi_3.2.xslt
>   https://github.com/LinuxCabal/admin-cfdi/pull/91/files#diff-8 (104)
> - _R_ admincfdi/bin/get_certificado.xslt
>   https://github.com/LinuxCabal/admin-cfdi/pull/91/files#diff-9 (4)
> - _R_ admincfdi/bin/get_sello.xslt
>   https://github.com/LinuxCabal/admin-cfdi/pull/91/files#diff-10 (2)
> - _R_ admincfdi/bin/get_sello_sat.xslt
>   https://github.com/LinuxCabal/admin-cfdi/pull/91/files#diff-11 (0)
> - _R_ admincfdi/bin/iconv.dll
>   https://github.com/LinuxCabal/admin-cfdi/pull/91/files#diff-12 (0)
> - _R_ admincfdi/bin/libeay32.dll
>   https://github.com/LinuxCabal/admin-cfdi/pull/91/files#diff-13 (0)
> - _R_ admincfdi/bin/libexslt.dll
>   https://github.com/LinuxCabal/admin-cfdi/pull/91/files#diff-14 (0)
> - _R_ admincfdi/bin/libxml2.dll
>   https://github.com/LinuxCabal/admin-cfdi/pull/91/files#diff-15 (0)
> - _R_ admincfdi/bin/libxslt.dll
>   https://github.com/LinuxCabal/admin-cfdi/pull/91/files#diff-16 (0)
> - _R_ admincfdi/bin/openssl.exe
>   https://github.com/LinuxCabal/admin-cfdi/pull/91/files#diff-17 (0)
> - _R_ admincfdi/bin/ssleay32.dll
>   https://github.com/LinuxCabal/admin-cfdi/pull/91/files#diff-18 (0)
> - _R_ admincfdi/bin/timbre.xslt
>   https://github.com/LinuxCabal/admin-cfdi/pull/91/files#diff-19 (0)
> - _R_ admincfdi/bin/timbre_1.0.xslt
>   https://github.com/LinuxCabal/admin-cfdi/pull/91/files#diff-20 (0)
> - _R_ admincfdi/bin/xsltproc.exe
>   https://github.com/LinuxCabal/admin-cfdi/pull/91/files#diff-21 (0)
> - _R_ admincfdi/bin/zlib1.dll
>   https://github.com/LinuxCabal/admin-cfdi/pull/91/files#diff-22 (0)
> - _R_ admincfdi/cer_pac/00001000000103834451.cer
>   https://github.com/LinuxCabal/admin-cfdi/pull/91/files#diff-23 (0)
> - _R_ admincfdi/cer_pac/00001000000104731997.cer
>   https://github.com/LinuxCabal/admin-cfdi/pull/91/files#diff-24 (0)
> - _R_ admincfdi/cer_pac/00001000000104750010.cer
>   https://github.com/LinuxCabal/admin-cfdi/pull/91/files#diff-25 (0)
> - _R_ admincfdi/cer_pac/00001000000104871381.cer
>   https://github.com/LinuxCabal/admin-cfdi/pull/91/files#diff-26 (0)
> - _R_ admincfdi/cer_pac/00001000000104888042.cer
>   https://github.com/LinuxCabal/admin-cfdi/pull/91/files#diff-27 (0)
> - _R_ admincfdi/cer_pac/00001000000200005634.cer
>   https://github.com/LinuxCabal/admin-cfdi/pull/91/files#diff-28 (0)
> - _R_ admincfdi/cer_pac/00001000000200011997.cer
>   https://github.com/LinuxCabal/admin-cfdi/pull/91/files#diff-29 (0)
> - _R_ admincfdi/cer_pac/00001000000200365214.cer
>   https://github.com/LinuxCabal/admin-cfdi/pull/91/files#diff-30 (0)
> - _R_ admincfdi/cer_pac/00001000000200795916.cer
>   https://github.com/LinuxCabal/admin-cfdi/pull/91/files#diff-31 (0)
> - _R_ admincfdi/cer_pac/00001000000201345662.cer
>   https://github.com/LinuxCabal/admin-cfdi/pull/91/files#diff-32 (0)
> - _R_ admincfdi/cer_pac/00001000000201345708.cer
>   https://github.com/LinuxCabal/admin-cfdi/pull/91/files#diff-33 (0)
> - _R_ admincfdi/cer_pac/00001000000201395217.cer
>   https://github.com/LinuxCabal/admin-cfdi/pull/91/files#diff-34 (0)
> - _R_ admincfdi/cer_pac/00001000000201455572.cer
>   https://github.com/LinuxCabal/admin-cfdi/pull/91/files#diff-35 (0)
> - _R_ admincfdi/cer_pac/00001000000201614141.cer
>   https://github.com/LinuxCabal/admin-cfdi/pull/91/files#diff-36 (0)
> - _R_ admincfdi/cer_pac/00001000000201629292.cer
>   https://github.com/LinuxCabal/admin-cfdi/pull/91/files#diff-37 (0)
> - _R_ admincfdi/cer_pac/00001000000201748120.cer
>   https://github.com/LinuxCabal/admin-cfdi/pull/91/files#diff-38 (0)
> - _R_ admincfdi/cer_pac/00001000000202241710.cer
>   https://github.com/LinuxCabal/admin-cfdi/pull/91/files#diff-39 (0)
> - _R_ admincfdi/cer_pac/00001000000202453260.cer
>   https://github.com/LinuxCabal/admin-cfdi/pull/91/files#diff-40 (0)
> - _R_ admincfdi/cer_pac/00001000000202638162.cer
>   https://github.com/LinuxCabal/admin-cfdi/pull/91/files#diff-41 (0)
> - _R_ admincfdi/cer_pac/00001000000202639096.cer
>   https://github.com/LinuxCabal/admin-cfdi/pull/91/files#diff-42 (0)
> - _R_ admincfdi/cer_pac/00001000000202639521.cer
>   https://github.com/LinuxCabal/admin-cfdi/pull/91/files#diff-43 (0)
> - _R_ admincfdi/cer_pac/00001000000202693892.cer
>   https://github.com/LinuxCabal/admin-cfdi/pull/91/files#diff-44 (0)
> - _R_ admincfdi/cer_pac/00001000000202695775.cer
>   https://github.com/LinuxCabal/admin-cfdi/pull/91/files#diff-45 (0)
> - _R_ admincfdi/cer_pac/00001000000202700691.cer
>   https://github.com/LinuxCabal/admin-cfdi/pull/91/files#diff-46 (0)
> - _R_ admincfdi/cer_pac/00001000000202771790.cer
>   https://github.com/LinuxCabal/admin-cfdi/pull/91/files#diff-47 (0)
> - _R_ admincfdi/cer_pac/00001000000202772539.cer
>   https://github.com/LinuxCabal/admin-cfdi/pull/91/files#diff-48 (0)
> - _R_ admincfdi/cer_pac/00001000000202809550.cer
>   https://github.com/LinuxCabal/admin-cfdi/pull/91/files#diff-49 (0)
> - _R_ admincfdi/cer_pac/00001000000202864285.cer
>   https://github.com/LinuxCabal/admin-cfdi/pull/91/files#diff-50 (0)
> - _R_ admincfdi/cer_pac/00001000000202864530.cer
>   https://github.com/LinuxCabal/admin-cfdi/pull/91/files#diff-51 (0)
> - _R_ admincfdi/cer_pac/00001000000202864883.cer
>   https://github.com/LinuxCabal/admin-cfdi/pull/91/files#diff-52 (0)
> - _R_ admincfdi/cer_pac/00001000000202865018.cer
>   https://github.com/LinuxCabal/admin-cfdi/pull/91/files#diff-53 (0)
> - _R_ admincfdi/cer_pac/00001000000202905407.cer
>   https://github.com/LinuxCabal/admin-cfdi/pull/91/files#diff-54 (0)
> - _R_ admincfdi/cer_pac/00001000000203015571.cer
>   https://github.com/LinuxCabal/admin-cfdi/pull/91/files#diff-55 (0)
> - _R_ admincfdi/cer_pac/00001000000203051706.cer
>   https://github.com/LinuxCabal/admin-cfdi/pull/91/files#diff-56 (0)
> - _R_ admincfdi/cer_pac/00001000000203082087.cer
>   https://github.com/LinuxCabal/admin-cfdi/pull/91/files#diff-57 (0)
> - _R_ admincfdi/cer_pac/00001000000203092957.cer
>   https://github.com/LinuxCabal/admin-cfdi/pull/91/files#diff-58 (0)
> - _R_ admincfdi/cer_pac/00001000000203093174.cer
>   https://github.com/LinuxCabal/admin-cfdi/pull/91/files#diff-59 (0)
> - _R_ admincfdi/cer_pac/00001000000203159220.cer
>   https://github.com/LinuxCabal/admin-cfdi/pull/91/files#diff-60 (0)
> - _R_ admincfdi/cer_pac/00001000000203159375.cer
>   https://github.com/LinuxCabal/admin-cfdi/pull/91/files#diff-61 (0)
> - _R_ admincfdi/cer_pac/00001000000203191015.cer
>   https://github.com/LinuxCabal/admin-cfdi/pull/91/files#diff-62 (0)
> - _R_ admincfdi/cer_pac/00001000000203220518.cer
>   https://github.com/LinuxCabal/admin-cfdi/pull/91/files#diff-63 (0)
> - _R_ admincfdi/cer_pac/00001000000203220546.cer
>   https://github.com/LinuxCabal/admin-cfdi/pull/91/files#diff-64 (0)
> - _R_ admincfdi/cer_pac/00001000000203253077.cer
>   https://github.com/LinuxCabal/admin-cfdi/pull/91/files#diff-65 (0)
> - _R_ admincfdi/cer_pac/00001000000203285726.cer
>   https://github.com/LinuxCabal/admin-cfdi/pull/91/files#diff-66 (0)
> - _R_ admincfdi/cer_pac/00001000000203285735.cer
>   https://github.com/LinuxCabal/admin-cfdi/pull/91/files#diff-67 (0)
> - _R_ admincfdi/cer_pac/00001000000203292609.cer
>   https://github.com/LinuxCabal/admin-cfdi/pull/91/files#diff-68 (0)
> - _R_ admincfdi/cer_pac/00001000000203312933.cer
>   https://github.com/LinuxCabal/admin-cfdi/pull/91/files#diff-69 (0)
> - _R_ admincfdi/cer_pac/00001000000203352843.cer
>   https://github.com/LinuxCabal/admin-cfdi/pull/91/files#diff-70 (0)
> - _R_ admincfdi/cer_pac/00001000000203392777.cer
>   https://github.com/LinuxCabal/admin-cfdi/pull/91/files#diff-71 (0)
> - _R_ admincfdi/cer_pac/00001000000203430011.cer
>   https://github.com/LinuxCabal/admin-cfdi/pull/91/files#diff-72 (0)
> - _R_ admincfdi/cer_pac/00001000000203495276.cer
>   https://github.com/LinuxCabal/admin-cfdi/pull/91/files#diff-73 (0)
> - _R_ admincfdi/cer_pac/00001000000203495475.cer
>   https://github.com/LinuxCabal/admin-cfdi/pull/91/files#diff-74 (0)
> - _R_ admincfdi/cer_pac/00001000000203631919.cer
>   https://github.com/LinuxCabal/admin-cfdi/pull/91/files#diff-75 (0)
> - _R_ admincfdi/cer_pac/00001000000300091673.cer
>   https://github.com/LinuxCabal/admin-cfdi/pull/91/files#diff-76 (0)
> - _R_ admincfdi/cer_pac/00001000000300171291.cer
>   https://github.com/LinuxCabal/admin-cfdi/pull/91/files#diff-77 (0)
> - _R_ admincfdi/cer_pac/00001000000300171326.cer
>   https://github.com/LinuxCabal/admin-cfdi/pull/91/files#diff-78 (0)
> - _R_ admincfdi/cer_pac/00001000000300209963.cer
>   https://github.com/LinuxCabal/admin-cfdi/pull/91/files#diff-79 (0)
> - _R_ admincfdi/cer_pac/00001000000300250292.cer
>   https://github.com/LinuxCabal/admin-cfdi/pull/91/files#diff-80 (0)
> - _R_ admincfdi/cer_pac/00001000000300392385.cer
>   https://github.com/LinuxCabal/admin-cfdi/pull/91/files#diff-81 (0)
> - _R_ admincfdi/cer_pac/00001000000300407877.cer
>   https://github.com/LinuxCabal/admin-cfdi/pull/91/files#diff-82 (0)
> - _R_ admincfdi/cer_pac/00001000000300439968.cer
>   https://github.com/LinuxCabal/admin-cfdi/pull/91/files#diff-83 (0)
> - _R_ admincfdi/cer_pac/00001000000300494998.cer
>   https://github.com/LinuxCabal/admin-cfdi/pull/91/files#diff-84 (0)
> - _R_ admincfdi/cer_pac/00001000000300627194.cer
>   https://github.com/LinuxCabal/admin-cfdi/pull/91/files#diff-85 (0)
> - _R_ admincfdi/cer_pac/00001000000300716418.cer
>   https://github.com/LinuxCabal/admin-cfdi/pull/91/files#diff-86 (0)
> - _R_ admincfdi/cer_pac/00001000000300716428.cer
>   https://github.com/LinuxCabal/admin-cfdi/pull/91/files#diff-87 (0)
> - _R_ admincfdi/cer_pac/00001000000300774022.cer
>   https://github.com/LinuxCabal/admin-cfdi/pull/91/files#diff-88 (0)
> - _R_ admincfdi/cer_pac/00001000000300915978.cer
>   https://github.com/LinuxCabal/admin-cfdi/pull/91/files#diff-89 (0)
> - _R_ admincfdi/cer_pac/00001000000300969660.cer
>   https://github.com/LinuxCabal/admin-cfdi/pull/91/files#diff-90 (0)
> - _R_ admincfdi/cer_pac/00001000000301021501.cer
>   https://github.com/LinuxCabal/admin-cfdi/pull/91/files#diff-91 (0)
> - _R_ admincfdi/cer_pac/00001000000301032322.cer
>   https://github.com/LinuxCabal/admin-cfdi/pull/91/files#diff-92 (0)
> - _R_ admincfdi/cer_pac/00001000000301062628.cer
>   https://github.com/LinuxCabal/admin-cfdi/pull/91/files#diff-93 (0)
> - _R_ admincfdi/cer_pac/00001000000301083052.cer
>   https://github.com/LinuxCabal/admin-cfdi/pull/91/files#diff-94 (0)
> - _R_ admincfdi/cer_pac/00001000000301100488.cer
>   https://github.com/LinuxCabal/admin-cfdi/pull/91/files#diff-95 (0)
> - _R_ admincfdi/cer_pac/00001000000301160463.cer
>   https://github.com/LinuxCabal/admin-cfdi/pull/91/files#diff-96 (0)
> - _A_ admincfdi/cer_pac/00001000000301205071.cer
>   https://github.com/LinuxCabal/admin-cfdi/pull/91/files#diff-97 (0)
> - _A_ admincfdi/cer_pac/00001000000301251152.cer
>   https://github.com/LinuxCabal/admin-cfdi/pull/91/files#diff-98 (0)
> - _R_ admincfdi/cer_pac/00001000000301280594.cer
>   https://github.com/LinuxCabal/admin-cfdi/pull/91/files#diff-99 (0)
> - _R_ admincfdi/cer_pac/00001000000301567711.cer
>   https://github.com/LinuxCabal/admin-cfdi/pull/91/files#diff-100 (0)
> - _R_ admincfdi/cer_pac/00001000000301634628.cer
>   https://github.com/LinuxCabal/admin-cfdi/pull/91/files#diff-101 (0)
> - _R_ admincfdi/cer_pac/00001000000301751173.cer
>   https://github.com/LinuxCabal/admin-cfdi/pull/91/files#diff-102 (0)
> - _R_ admincfdi/cer_pac/00001000000301914249.cer
>   https://github.com/LinuxCabal/admin-cfdi/pull/91/files#diff-103 (0)
> - _R_ admincfdi/cer_pac/00001000000301927035.cer
>   https://github.com/LinuxCabal/admin-cfdi/pull/91/files#diff-104 (0)
> - _R_ admincfdi/cer_pac/00001000000301949314.cer
>   https://github.com/LinuxCabal/admin-cfdi/pull/91/files#diff-105 (0)
> - _R_ admincfdi/cer_pac/00001000000304339685.cer
>   https://github.com/LinuxCabal/admin-cfdi/pull/91/files#diff-106 (0)
> - _R_ admincfdi/cer_pac/00001000000304691381.cer
>   https://github.com/LinuxCabal/admin-cfdi/pull/91/files#diff-107 (0)
> - _R_ admincfdi/cer_pac/20001000000100005868.cer
>   https://github.com/LinuxCabal/admin-cfdi/pull/91/files#diff-108 (0)
> - _R_ admincfdi/img/calc.gif
>   https://github.com/LinuxCabal/admin-cfdi/pull/91/files#diff-109 (0)
> - _R_ admincfdi/img/calc.png
>   https://github.com/LinuxCabal/admin-cfdi/pull/91/files#diff-110 (0)
> - _R_ admincfdi/img/csv.gif
>   https://github.com/LinuxCabal/admin-cfdi/pull/91/files#diff-111 (0)
> - _R_ admincfdi/img/csv.png
>   https://github.com/LinuxCabal/admin-cfdi/pull/91/files#diff-112 (0)
> - _R_ admincfdi/img/delete.gif
>   https://github.com/LinuxCabal/admin-cfdi/pull/91/files#diff-113 (0)
> - _R_ admincfdi/img/delete.png
>   https://github.com/LinuxCabal/admin-cfdi/pull/91/files#diff-114 (0)
> - _R_ admincfdi/img/down.gif
>   https://github.com/LinuxCabal/admin-cfdi/pull/91/files#diff-115 (0)
> - _R_ admincfdi/img/down.png
>   https://github.com/LinuxCabal/admin-cfdi/pull/91/files#diff-116 (0)
> - _R_ admincfdi/img/exit.gif
>   https://github.com/LinuxCabal/admin-cfdi/pull/91/files#diff-117 (0)
> - _R_ admincfdi/img/exit.png
>   https://github.com/LinuxCabal/admin-cfdi/pull/91/files#diff-118 (0)
> - _R_ admincfdi/img/favicon.gif
>   https://github.com/LinuxCabal/admin-cfdi/pull/91/files#diff-119 (0)
> - _R_ admincfdi/img/favicon.png
>   https://github.com/LinuxCabal/admin-cfdi/pull/91/files#diff-120 (0)
> - _R_ admincfdi/img/folder.gif
>   https://github.com/LinuxCabal/admin-cfdi/pull/91/files#diff-121 (0)
> - _R_ admincfdi/img/folder.png
>   https://github.com/LinuxCabal/admin-cfdi/pull/91/files#diff-122 (0)
> - _R_ admincfdi/img/pdf.gif
>   https://github.com/LinuxCabal/admin-cfdi/pull/91/files#diff-123 (0)
> - _R_ admincfdi/img/pdf.png
>   https://github.com/LinuxCabal/admin-cfdi/pull/91/files#diff-124 (0)
> - _R_ admincfdi/img/report.gif
>   https://github.com/LinuxCabal/admin-cfdi/pull/91/files#diff-125 (0)
> - _R_ admincfdi/img/report.png
>   https://github.com/LinuxCabal/admin-cfdi/pull/91/files#diff-126 (0)
> - _R_ admincfdi/img/save.gif
>   https://github.com/LinuxCabal/admin-cfdi/pull/91/files#diff-127 (0)
> - _R_ admincfdi/img/save.png
>   https://github.com/LinuxCabal/admin-cfdi/pull/91/files#diff-128 (0)
> - _R_ admincfdi/img/xml.gif
>   https://github.com/LinuxCabal/admin-cfdi/pull/91/files#diff-129 (0)
> - _R_ admincfdi/img/xml.png
>   https://github.com/LinuxCabal/admin-cfdi/pull/91/files#diff-130 (0)
> - _R_ admincfdi/pyutil.py
>   https://github.com/LinuxCabal/admin-cfdi/pull/91/files#diff-131
>   (1223)
> - _A_ admincfdi/template/default.csv
>   https://github.com/LinuxCabal/admin-cfdi/pull/91/files#diff-132 (77)
> - _A_ admincfdi/template/plantilla_factura.ods
>   https://github.com/LinuxCabal/admin-cfdi/pull/91/files#diff-133 (0)
> - _A_ admincfdi/tests/test_pyutil.py
>   https://github.com/LinuxCabal/admin-cfdi/pull/91/files#diff-134
>   (261)
> - _R_ admincfdi/ui/mainwindow.ui
>   https://github.com/LinuxCabal/admin-cfdi/pull/91/files#diff-135 (12)
> - _R_ admincfdi/values.py
>   https://github.com/LinuxCabal/admin-cfdi/pull/91/files#diff-136 (17)
> - _A_ cfdi2pdf
>   https://github.com/LinuxCabal/admin-cfdi/pull/91/files#diff-137 (56)
> - _A_ cfdi2pdf.cmd
>   https://github.com/LinuxCabal/admin-cfdi/pull/91/files#diff-138 (4)
> - _A_ contributors.txt
>   https://github.com/LinuxCabal/admin-cfdi/pull/91/files#diff-139 (5)
> - _A_ descarga-cfdi
>   https://github.com/LinuxCabal/admin-cfdi/pull/91/files#diff-140
>   (129)
> - _A_ descarga-cfdi.cmd
>   https://github.com/LinuxCabal/admin-cfdi/pull/91/files#diff-141 (4)
> - _M_ docs/conf.py
>   https://github.com/LinuxCabal/admin-cfdi/pull/91/files#diff-142 (8)
> - _M_ docs/devel.rst
>   https://github.com/LinuxCabal/admin-cfdi/pull/91/files#diff-143
>   (309)
> - _M_ docs/glosario.rst
>   https://github.com/LinuxCabal/admin-cfdi/pull/91/files#diff-144 (2)
> - _A_ docs/img/admin-cfdi-ventana-ppal.png
>   https://github.com/LinuxCabal/admin-cfdi/pull/91/files#diff-145 (0)
> - _A_ docs/img/ejemplo-pdf.png
>   https://github.com/LinuxCabal/admin-cfdi/pull/91/files#diff-146 (0)
> - _M_ docs/index.rst
>   https://github.com/LinuxCabal/admin-cfdi/pull/91/files#diff-147 (2)
> - _A_ docs/install.rst
>   https://github.com/LinuxCabal/admin-cfdi/pull/91/files#diff-148 (66)
> - _M_ docs/intro.rst
>   https://github.com/LinuxCabal/admin-cfdi/pull/91/files#diff-149 (79)
> - _A_ docs/reference.rst
>   https://github.com/LinuxCabal/admin-cfdi/pull/91/files#diff-150 (27)
> - _M_ docs/uso.rst
>   https://github.com/LinuxCabal/admin-cfdi/pull/91/files#diff-151
>   (128)
> - _A_ functional_DescargaSAT.conf.sample
>   https://github.com/LinuxCabal/admin-cfdi/pull/91/files#diff-152 (21)
> - _A_ functional_DescargaSAT.py
>   https://github.com/LinuxCabal/admin-cfdi/pull/91/files#diff-153
>   (213)
> - _A_ pwd.sample
>   https://github.com/LinuxCabal/admin-cfdi/pull/91/files#diff-154 (3)
> - _D_ pyqrcode/**init**.py
>   https://github.com/LinuxCabal/admin-cfdi/pull/91/files#diff-155
>   (322)
> - _D_ pyqrcode/builder.py
>   https://github.com/LinuxCabal/admin-cfdi/pull/91/files#diff-156
>   (1089)
> - _D_ pyqrcode/png.py
>   https://github.com/LinuxCabal/admin-cfdi/pull/91/files#diff-157
>   (3838)
> - _D_ pyqrcode/tables.py
>   https://github.com/LinuxCabal/admin-cfdi/pull/91/files#diff-158
>   (746)
> - _D_ pysimplesoap/**init**.py
>   https://github.com/LinuxCabal/admin-cfdi/pull/91/files#diff-159 (16)
> - _D_ pysimplesoap/client.py
>   https://github.com/LinuxCabal/admin-cfdi/pull/91/files#diff-160
>   (713)
> - _D_ pysimplesoap/helpers.py
>   https://github.com/LinuxCabal/admin-cfdi/pull/91/files#diff-161
>   (465)
> - _D_ pysimplesoap/server.py
>   https://github.com/LinuxCabal/admin-cfdi/pull/91/files#diff-162
>   (560)
> - _D_ pysimplesoap/simplexml.py
>   https://github.com/LinuxCabal/admin-cfdi/pull/91/files#diff-163
>   (481)
> - _D_ pysimplesoap/transport.py
>   https://github.com/LinuxCabal/admin-cfdi/pull/91/files#diff-164
>   (274)
> - _A_ requirements.txt
>   https://github.com/LinuxCabal/admin-cfdi/pull/91/files#diff-165 (6)
> - _A_ setup.py
>   https://github.com/LinuxCabal/admin-cfdi/pull/91/files#diff-166 (27)
> - _D_ template.py
>   https://github.com/LinuxCabal/admin-cfdi/pull/91/files#diff-167
>   (176)
> 
> Patch Links:
> - https://github.com/LinuxCabal/admin-cfdi/pull/91.patch
> - https://github.com/LinuxCabal/admin-cfdi/pull/91.diff
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/LinuxCabal/admin-cfdi/pull/91.

## 

http://pixelab.com.mx Carlos Omar Briseño Gutierrez
Software Engineering
Cell:(214) 797-7702
omar.b@pixelab.com.mx
http://pixelab.com.mx
_f_ https://www.facebook.com/carlos.o.gutierrez _t_
https://twitter.com/omar7c1

ghost commented:
Creo que si la página tiene soporte para hora, minuto y segundo, lo más sencillo es solo trasladar dicho soporte y que el emisor decida en que rangos los usa... ¿no creen?

Excelente... ¿mezclamos?...

ppaez commented:
Ok, incluiremos las opciones para especificar segundo_inicial y segundo_final, con valores por omisión 0 y 59.  Todavía no mezclar... aún faltan cambios, aviso cuando estén listos.

ppaez committed:
Soportar hora y minuto inicial

- Agregar parámetros opcionales hora_inicial y
  minuto_inicial a DescargaSAT.search()
- Agregar una prueba funcional
- Agregar ambos parámetros opcionales a descarga-cfdi
- admin-cfdi y descarga-cfdi pasan ambos parámetros
  a DescargaSAT.search()

ppaez committed:
Soportar segundo inicial y final

- Agregar parámetros opcionales segundo_inicial y
  segundo_final a DescargaSAT.search()
- Agregar una prueba funcional
- Agregar ambos parámetros opcionales a descarga-cfdi
- admin-cfdi y descarga-cfdi pasan ambos parámetros
  a DescargaSAT.search()

ppaez committed:
Actualizar la documentación

ppaez committed:
Correción al remover ceros

- El método anterior convierte '0' en ''
- Cubrir con prueba unitaria

ppaez commented:
Listos los cambios adicionales, las pruebas funcionales pasan en Debian y W10.

ghost commented:
Excelente...
